core/server/master/src/main/java/alluxio/underfs/UfsStatusCache.java

```diff
@@ -210,22 +213,28 @@ public UfsStatus fetchStatusIfAbsent(AlluxioURI path, MountTable mountTable)
    * @throws InvalidPathException if the alluxio path can't be resolved to a UFS mount
    */
   @Nullable
-  public Collection<UfsStatus> fetchChildrenIfAbsent(AlluxioURI path, MountTable mountTable,
-      boolean useFallback)
+  public Collection<UfsStatus> fetchChildrenIfAbsent(RpcContext rpcContext, AlluxioURI path,
+                                                     MountTable mountTable, boolean useFallback)
       throws InterruptedException, InvalidPathException {
     Future<Collection<UfsStatus>> prefetchJob = mActivePrefetchJobs.get(path);
     if (prefetchJob != null) {
-      try {
-        return prefetchJob.get();
-      } catch (InterruptedException | ExecutionException e) {
-        LogUtils.warnWithException(LOG, "Failed to get result for prefetch job on alluxio path {}",
-            path, e);
-        if (e instanceof InterruptedException) {
-          Thread.currentThread().interrupt();
-          throw (InterruptedException) e;
+      while (true) {
+        try {
+          return prefetchJob.get(100, TimeUnit.MILLISECONDS);
+        } catch (TimeoutException e) {
+          if (rpcContext != null) {
+            rpcContext.throwIfCancelled();
```

Comment:
is there a way we can test the cancellation and how our system responds to a runtimeException being thrown in the middle of a sync?